pkg/build/api/types.go

```diff
@@ -397,6 +397,9 @@ const (
 	// availability of a new version of an image
 	ImageChangeBuildTriggerType           BuildTriggerType = "ImageChange"
 	ImageChangeBuildTriggerTypeDeprecated BuildTriggerType = "imageChange"
+
+	// ConfigChangeBuildTriggerType will trigger a build on an initial build config creation
+	ConfigChangeBuildTriggerType BuildTriggerType = "ConfigChange"
```

Comment:
I'd add clear statement in the docs to inform users this behavior *will change*. Although I tend to agree with @liggitt about unexpected modification in the future, on the other hand I don't want to mess out our API with two distinct objects one for initial config change and additional one for subsequent changes.